src/IB/IBFEMethod.cpp

```diff
@@ -1362,8 +1362,15 @@ void IBFEMethod::endDataRedistribution(Pointer<PatchHierarchy<NDIM> > /*hierarch
             // TODO: d_current_time is actually nan here. Since we don't do
             // any sort of tagging based on the current time I think we can
             // just put in a bogus (nonnegative) value.
+            int max_level = d_scratch_hierarchy->getFinestLevelNumber();
+            if (max_level == 0)
+              {
+                TBOX_ERROR("IBFEMethod::endDataRedistribution()\n"
+                           << "At the current time the scratch hierarchy is only available\n"
+                           << "with AMR (i.e., there must be at least two levels.)\n");
+              }
             d_scratch_gridding_algorithm->regridAllFinerLevels(
-                d_scratch_hierarchy, 0, 0.0 /*d_current_time*/, tag_buffer);
+                d_scratch_hierarchy, max_level - 1, 0.0 /*d_current_time*/, tag_buffer);
```

Comment:
To get this to work with a single level, wouldn't this do the trick:
```suggestion
                d_scratch_hierarchy, std::max(0, max_level - 1), 0.0 /*d_current_time*/, tag_buffer);
```

Reply:
I don't think so because if there is only one level it doesn't make sense to regrid all finer levels.

Reply:
Can you give it a try? I think this is how patches are distributed for single-level runs.

Reply:
I'm on it!

Reply:
Ah, [here](https://ibamr.github.io/IBAMR-docs/samrai/2.4.4/html/classSAMRAI_1_1mesh_1_1GriddingAlgorithm.html#4dd69a9bf65bfe13b4cd6712e868c607) is the relevant documentation.

Reply:
Should use `GriddingAlgorithm::makeCoarsestLevel()` for a single-level run.

Reply:
you're right - surprisingly this works.

Reply:
As in, I am surprised that nothing else was broken in this case.